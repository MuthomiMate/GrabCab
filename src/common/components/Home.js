import React, {Component} from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';
import Splash from "./splash";


class Home extends Component {
  constructor(props){
    super(props)
    this.state = {
      showNextPage: false,
    }
  }

  componentDidMount(){
    setTimeout(()=> {
      this.setState( { showNextPage: true } )
    }, 5000)
  }

  render(){
    const { showNextPage } = this.state;
    return (
        <Splash
          style={styles.container}
          showButtons={showNextPage}
          navigation={this.props.navigation}
        />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
  },
});

export default Home;