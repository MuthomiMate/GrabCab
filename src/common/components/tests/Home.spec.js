import React from 'react';
import Home from '../Home';
import { shallow } from 'enzyme';
describe("<Home />", ()=> {

    it("renders correctly", () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot()
    });

    it("changes state after 5 seconds", (done) => {
        const wrapper = shallow(<Home />);
        setTimeout(()=> {
            expect(wrapper.state().showNextPage).toBe(true)
            done();
          }, 5000)
    })
})