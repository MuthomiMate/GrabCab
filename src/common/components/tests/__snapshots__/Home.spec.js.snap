// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<Splash
  showButtons={false}
  style={
    Object {
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
/>
`;
